### PR TITLE
Rework tx builder

### PR DESCRIPTION
**Pull request by robert-zaremba**
Updated transaction creation and parameters packing.

### CODE REVIEW

File: packages/algob/src/lib/tx.ts [hunk 1]
```diff
@@ -189,49 +192,43 @@ function signTransaction (txn: Transaction, txnParam: execParams): Uint8Array {
...
+    mkTransaction(p,
+      await mkTxParams(deployer.algodClient, p.payFlags, Object.assign({}, suggestedParams)));
+
+  let signedTxn;
+  if (Array.isArray(execParams)) {
+    if (execParams.length > 16) { throw new Error("Maximum size of an atomic transfer group is 16"); }
+
+    let txns = [];
+    for (const t of execParams) { txns.push(await mkTx(t)); }
+    txns = algosdk.assignGroupID(txns);
```
Comment (robert-zaremba) [inline packages/algob/src/lib/tx.ts:218]: here, it's safer to assign the result of `assignGroupID`.

File: packages/algob/src/lib/tx.ts [hunk 2]
```diff
@@ -189,49 +192,43 @@ function signTransaction (txn: Transaction, txnParam: execParams): Uint8Array {
...
+    signedTxn = txns.map((txn: Transaction, index: number) => {
```
Comment (amityadav0) [inline packages/algob/src/lib/tx.ts:219]: I am curious about this. does this creates a map and stores it in signedTxn ? How is it different than using `forEach` ?

**Review by amityadav0 — APPROVED** [review]
LGTM
tested atomic transaction example also.

File: packages/algob/src/lib/tx.ts (hunk 2 above)
Comment (robert-zaremba) [reply packages/algob/src/lib/tx.ts:219]: `map` returns a list. `forEach` only iterates
explanation of map:
```
let L = some_list;
L.map(f) == [f(L[0]), f(L[1]), .... f(L[n])]
```

**Pull request merged**